pkg/util/repospec.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2019 The Kubernetes Authors.
```

Comment:
Should we add a comment explaining where is this coming from?

Reply:
I was wondering if this copy-paste of kustomize code is even allowed, according to their license. Who can we ask about it?

Reply:
They use Apache2.0 liscense, we also use it. I think we should keep the copyright and we should be good: 
![image](https://user-images.githubusercontent.com/60569147/119335231-c74fa780-bc94-11eb-9082-08d383684d4a.png)

Reply:
added comments in code (like https://github.com/argoproj/argo-cd/blob/release-2.0/util/git/workaround.go#L13)